src/NServiceBus.AcceptanceTests/Config/When_startup_is_complete.cs

```diff
@@ -25,34 +26,48 @@ public class Context : ScenarioContext
             public bool SettingIsAvailable { get; set; }
         }
 
-        public class StartedEndpoint : EndpointConfigurationBuilder
+        class AfterConfigIsCompleteFeature : Feature
         {
-            public StartedEndpoint()
+            protected override void Setup(FeatureConfigurationContext context)
             {
-                EndpointSetup<DefaultServer>();
+                context.Container.ConfigureComponent<AfterConfigIsCompleteFeatureTask>(DependencyLifecycle.SingleInstance);
+                context.RegisterStartupTask(b => b.Build<AfterConfigIsCompleteFeatureTask>());
             }
 
-            class AfterConfigIsComplete:IWantToRunWhenBusStartsAndStops
+            class AfterConfigIsCompleteFeatureTask : FeatureStartupTask
             {
-                public Context Context { get; set; }
-
-                public ReadOnlySettings Settings { get; set; }
+                readonly ReadOnlySettings settings;
+                readonly Context context;
 
+                public AfterConfigIsCompleteFeatureTask(ReadOnlySettings settings, Context context)
+                {
+                    this.settings = settings;
+                    this.context = context;
+                }
 
-                public Task Start(IMessageSession session)
+                protected override Task OnStart(IMessageSession session)
                 {
-                    Context.SettingIsAvailable = Settings != null;
+                    context.SettingIsAvailable = settings != null;
 
-                    Context.IsDone = true;
+                    context.IsDone = true;
                     return Task.FromResult(0);
                 }
 
-                public Task Stop(IMessageSession session)
+                protected override Task OnStop(IMessageSession session)
                 {
                     return Task.FromResult(0);
                 }
             }
         }
+
+
```

Comment:
Yes
Am 21.03.2016 22:08 schrieb "Kyle Baley" <notifications@github.com>:

> In src/NServiceBus.AcceptanceTests/Config/When_startup_is_complete.cs
> <https://github.com/Particular/NServiceBus/pull/3611#discussion_r56897643>
> :
>
> >                  {
> >                      return Task.FromResult(0);
> >                  }
> >              }
> >          }
> > +
> > +
>
> The code cleanup also cleans up the using statements and the tests seem
> to have a convention of making the NSB references explicit. E.g. it'll
> convert this:
>
> using NServiceBus.AcceptanceTesting;
> using NServiceBus.AcceptanceTests.EndpointTemplates;
>
> to this:
>
> using AcceptanceTesting;
> using EndpointTemplates;
>
> Doing so in these files will make them inconsistent with the remaining
> tests. Should I still apply the rules?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/3611/files/b032f0695fcf5c697946845e7143f182144f1020#r56897643>
>